model/dependency.go

```diff
@@ -232,7 +256,11 @@ func (di *dependencyIncluder) expandDependencies(pair TVPair, depends []TaskUnit
 						if projectTask.IsDisabled() || projectTask.SkipOnRequester(di.requester) {
 							continue
 						}
-						deps = append(deps, TVPair{TaskName: t.Name, Variant: v.Name})
+						if !t.IsGroup {
+							deps = append(deps, TVPair{TaskName: t.Name, Variant: v.Name})
+						} else {
+							deps = append(deps, TVPair{TaskName: d.Name, Variant: v.Name})
+						}
```

Comment:
I found a small bug where it would add the whole task group rather than the particular dependency. This changes it to only add the particular task group task rather than the whole task group.

If it is a single host task group, the new logic will catch that (I also added a test catching that)